src/components/TextInput.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Field } from 'redux-form';
+import TextField from 'material-ui/TextField';
+import PropTypes from 'prop-types';
+
+function RenderTextField({ ...props }) {
+  const textProp = Object.assign({}, props);
+  delete textProp.meta;
+  delete textProp.input;
+
+  return (
+    <TextField
+      {...textProp}
+    />
+  );
+}
+const TextInput = ({ name, ...props }) => (
+  <Field
+    {...props}
+    name={name}
+    id={name}
```

Comment:
I get a complaint from Material-UI text saying it needs an id to function properly, i imagine that it would work fine without assigning an id a Field in this case takes care of that, if you agree with that then id can be taken off